xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/template/InternalTemplateManager.java

```diff
@@ -894,4 +895,20 @@ public Template getTemplate(String templateName)
 
         return template;
     }
+
+    /**
+     * Create a DefaultTemplate with the given content and the given author.
+     *
+     * @param content the template content
+     * @param author the template author
+     * @return the template
+     */
+    public Template createStringTemplate(String content, DocumentReference author)
+    {
+        DefaultTemplate newTemplate = new DefaultTemplate(null);
```

Comment:
> I don’t understand what you mean by generating a proper StringResource. How does using an AbstractInputSourceResource helps me to set a template content ?

If you look at how other templates are working you will see that the content come from the Resource. The Resource is then parsed to extract various properties (like the actual syntax used by the template), see http://extensions.xwiki.org/xwiki/bin/view/Extension/Template+Module#HContent for the currently supported properties.

What you are doing here is totally bypassing all this process to set what is supposed to be the final content after the parsing which is exactly what you where already doing in your previous public StringTemplate and that I did not liked.

Also are you sure you tested this code ? From what I see you should end up with a nullpoint  because "newTemplate.content" should be null (but it's possible I missed something).

>  Also, I thought that using a StringTemplate was useless ? #570 (comment)

I never said that. I said that having a public one not having all the features of AbtractTemplate was not very nice.

Here I'm suggesting you to have an internal StringTemplate which deal with the generation of the proper Resource to pass to AbstractTemplate. So just a constructor, that's all. I'm not talking about API but about code readability.